src/nprintml/label/aggregator/pcap.py

```diff
@@ -48,93 +55,113 @@ def normalize_npt(cls, npt_csv, path_input_base=None):
                 f"data result but stream was empty: {npt_csv}"
             )
 
+    @staticmethod
+    @storeresults
+    def flatten_npt(indexed_files):
+        print('Loading nPrints')
+
+        # We need every sample size to be the same, so we'll find the
+        # largest nPrint in all of the files and pad each sample with
+        # empty nPrints.
+        header = usecols = None
+        max_length = file_count = 0
+
+        for (file_count, (file_index, npt_file)) in enumerate(indexed_files, 1):
+            if header is None:
+                header = np.genfromtxt(npt_file, delimiter=',', max_rows=1, dtype=str)
+                usecols = np.arange(1, len(header))  # ignore data index (src_ip)
+                skip_header = 1 if isinstance(npt_file, (str, pathlib.Path)) else 0
+            else:
+                skip_header = 1
+
+            npt = np.genfromtxt(
+                npt_file,
+                delimiter=',',
+                skip_header=skip_header,
+                usecols=usecols,
+                dtype=NPT_DTYPE,
+            )
+
+            npt_dim1 = len(npt.shape) == 1
+            npt_length = 1 if npt_dim1 else len(npt)
+            max_length = max(max_length, npt_length)
+
+            npt_flat = npt if npt_dim1 else npt.ravel()
+
+            yield list(itertools.chain([file_index], npt_flat))
+
+        print('Loaded', file_count, 'nprints')
+
+        return (header[1:], max_length)
+
+    @classmethod
+    def merge_npt(cls, npts_flat):
+        """Merge nPrint data from multiple output files."""
+        npts = pd.DataFrame(npts_flat, dtype=NPT_DTYPE)
```

Comment:
Though the data is in fact now heterogeneous (str + int), Pandas appears to respect our dtype well enough :shrug: 

As far as I can tell there's really no way to specify our index and choice of `na` here; though, it's as yet unclear that we're really so badly bitten by having to specify these subsequently below.

Again, it's possible that we get something out of experimenting with this more … though that should most likely be left up to future work (and up to future interest).

Reply:
Yeah this likely should be a separate issue marked to look at. It may come up as fruitless, but it would seem weird that we can't supply our own NaN values.

Reply:
Sure. Would also make sense to confirm whether it amounts to anything &ndash; I was also hoping to avoid it, but it's entirely possible that it's imperceptible relative to the rest. (I'm sure AutoGluon commits worse sins, for one.)

Reply:
We now no longer attempt to set a dtype here (since it's really not possible and ineffective to specify). (See previous comment re: dtype.)